modules/terraform/count_test.go

```diff
@@ -0,0 +1,95 @@
+package terraform
+
+import (
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/files"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetResourceCount(t *testing.T) {
+	t.Parallel()
+	testFolder, err := files.CopyTerraformFolderToTemp("../../test/fixtures/terraform-basic-configuration", t.Name())
+	require.NoError(t, err)
+
+	terraformOptions := &Options{
+		TerraformDir: testFolder,
+		Vars: map[string]interface{}{
+			"cnt": 1,
+		},
+	}
+
+	cnt := GetResourceCount(t, InitAndPlan(t, terraformOptions))
+	assert.Equal(t, 1, cnt.Add)
+	assert.Equal(t, 0, cnt.Change)
+	assert.Equal(t, 0, cnt.Destroy)
+}
+
+func TestGetResourceCountEColor(t *testing.T) {
+	t.Parallel()
+	runTestGetResourceCountE(t, false)
+}
+
+func TestGetResourceCountENoColor(t *testing.T) {
+	t.Parallel()
+	runTestGetResourceCountE(t, true)
+}
+
+func runTestGetResourceCountE(t *testing.T, noColor bool) {
+	testCases := []struct {
+		Name                                         string
+		tfFuncToRun                                  func(t *testing.T, options *Options) string
+		cntValue                                     int
+		expectedAdd, expectedChange, expectedDestroy int
+	}{
+		{"Plan_Zero", InitAndPlan, 0, 0, 0, 0},
+		{"Apply_Zero", InitAndApply, 0, 0, 0, 0},
+		{"Plan_AddResouce", InitAndPlan, 2, 2, 0, 0},
+		{"Apply_AddResouce", InitAndApply, 2, 2, 0, 0},
+		{"Plan_NoOp", InitAndApply, 2, 0, 0, 0},
+		{"Apply_NoOp", InitAndApply, 2, 0, 0, 0},
+		{"Plan_DestroyResource", InitAndPlan, 1, 0, 0, 1},
+		{"Apply_DestroyResource", InitAndApply, 1, 0, 0, 1},
+		{"Destroy", Destroy, 1, 0, 0, 1},
+		{"Destroy_NoOp", Destroy, 1, 0, 0, 0},
+	}
+
+	notFailedSoFar := true
+
+	testFolder, err := files.CopyTerraformFolderToTemp("../../test/fixtures/terraform-basic-configuration", t.Name())
+	require.NoError(t, err)
+
+	terraformOptions := &Options{
+		TerraformDir: testFolder,
+		Vars: map[string]interface{}{
+			"cnt": 0,
+		},
+		NoColor: noColor,
+	}
+
+	for _, tc := range testCases {
+		if notFailedSoFar {
+			t.Run(tc.Name,
+				func(t *testing.T) {
+					terraformOptions.Vars["cnt"] = tc.cntValue
+					cnt, err := GetResourceCountE(t, tc.tfFuncToRun(t, terraformOptions))
+					notFailedSoFar = assert.NoError(t, err)
+					if notFailedSoFar {
+						notFailedSoFar = assert.Equal(t, tc.expectedAdd, cnt.Add) &&
+							assert.Equal(t, tc.expectedChange, cnt.Change) &&
+							assert.Equal(t, tc.expectedDestroy, cnt.Destroy)
+					}
+				})
+		}
+	}
+
+	if notFailedSoFar {
+		t.Run("InvalidInput",
+			func(t *testing.T) {
+				cnt, err := GetResourceCountE(t, "Invalid Terraform output")
```

Comment:
Can you come up with a way to make this test more realistic? E.g what if you pass in a string for `cnt` to the terraform module and have it error?

Reply:
done